### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'split'

### ISSUE THREAD

**Issue by domdfcoding** [joerick/pyinstrument#140]
Hi,

I've been trying out `pyinstrument`, but I'm getting an intermittent error:

```python
Traceback (most recent call last):
  File "venv/bin/pyinstrument", line 10, in <module>
    sys.exit(main())
  File "venv/lib/python3.8/site-packages/pyinstrument/__main__.py", line 307, in main
    f.write(renderer.render(session))
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/console.py", line 38, in render
    result += self.render_frame(self.root_frame)
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/console.py", line 111, in render_frame
    result += self.render_frame(child, indent=c_indent, child_indent=cc_indent)
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/console.py", line 111, in render_frame
    result += self.render_frame(child, indent=c_indent, child_indent=cc_indent)
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/console.py", line 111, in render_frame
    result += self.render_frame(child, indent=c_indent, child_indent=cc_indent)
  [Previous line repeated 10 more times]
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/console.py", line 92, in render_frame
    libraries=truncate(", ".join(frame.group.libraries), 40),
  File "venv/lib/python3.8/site-packages/pyinstrument/frame.py", line 408, in libraries
    library = frame.file_path_short.split(os.sep)[0]
AttributeError: 'NoneType' object has no attribute 'split'
```

If I re-run `pyinstrument` it completes successfully, but running again fails.

Running with `pdb` I can see that `frame` is:

```python
Frame(identifier=<module>1, time=0.001013, len(children)=1), group=FrameGroup(len(frames)=858)
```

This is the same error message (but not on the same line, I think) as #122

**Comment by joerick** [comment]
ahhh that makes sense, it's a <module> frame. Curious that I've never seen that before. Thanks for the report! Fix should be pretty easy...

**Comment by domdfcoding** [comment]
I managed to trigger it with HTML/JSON output as well:

```python
Traceback (most recent call last):
  File "venv/bin/pyinstrument", line 10, in <module>
    sys.exit(main())
  File "venv/lib/python3.8/site-packages/pyinstrument/__main__.py", line 304, in main
    output_filename = renderer.open_in_browser(session)
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/html.py", line 70, in open_in_browser
    f.write(self.render(session))
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/html.py", line 35, in render
    session_json = self.render_json(session)
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/html.py", line 83, in render_json
    return json_renderer.render(session)
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/jsonrenderer.py", line 64, in render
    property_decls.append('"root_frame": %s' % self.render_frame(frame))
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/jsonrenderer.py", line 44, in render_frame
    children_jsons.append(self.render_frame(child))
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/jsonrenderer.py", line 44, in render_frame
    children_jsons.append(self.render_frame(child))
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/jsonrenderer.py", line 44, in render_frame
    children_jsons.append(self.render_frame(child))
  [Previous line repeated 10 more times]
  File "venv/lib/python3.8/site-packages/pyinstrument/renderers/jsonrenderer.py", line 34, in render_frame
    property_decls.append('"file_path_short": %s' % encode_str(frame.file_path_short))
TypeError: first argument must be a string, not NoneType
```

Again the frame's identifier is `<module>1`

Only happens every 7th run this time (though this is on a different PC)

**Comment by joerick** [comment]
This should be fixed in v4.0.3! Let me know if not.

**Comment by domdfcoding** [comment]
Thanks! Despite trying I can't make it crash any more.
